app/src/main/java/org/eyeseetea/malariacare/data/net/ConnectivityManager.java

```diff
@@ -19,12 +19,8 @@ public ConnectivityManager(Context context) {
      */
     @Override
     public boolean isDeviceOnline() {
-        android.net.ConnectivityManager connMgr =
-                (android.net.ConnectivityManager) mContext.getSystemService(
-                        Context.CONNECTIVITY_SERVICE);
-
-        NetworkInfo networkInfo = connMgr.getNetworkInfo(android.net.ConnectivityManager.TYPE_WIFI);
-        return networkInfo.isConnected();
+        NetworkInfo networkInfo = getConnectivityNetwork();
```

Comment:
ConnectivityManager should not be modified on this issue, please merge with https://github.com/EyeSeeTea/pictureapp/pull/2247 branch again that has this changes and resolve conflicts